### ISSUE TITLE
Using i18n with beforeFiles is unpredictable

### ISSUE THREAD

**Issue by activenode** [vercel/next.js#45176]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.1.4
      eslint-config-next: 13.0.6
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

Internationalization (i18n)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/nextjs-i18n-issue-mrgkjh?file=%2Fnext.config.js

### To Reproduce

Define a `i18n` field in `next.config.js` as in the linked CodeSandbox or as described in  https://nextjs.org/docs/advanced-features/i18n-routing .

Try defining a rewrite in `beforeFiles` of the `next.config.js` such that any request to `/foo` goes to `/api/whatever` (in which case both are logically locale-less).

Now start your server and see that you cannot access `/foo` as it throws a `404`.

### Describe the Bug

tldr: It's all given in the reproduction description basically. 

It throws a `404` when defining a `rewrite` in `beforeFiles` in `next.config.js` when using a `i18n` definition and none of that is predictable / logical as per documentation.

### Expected Behavior

It should be very clear why this behaviour is happening - if it is intentional (nothing to be found here currently https://nextjs.org/docs/api-reference/next.config.js/rewrites).

Also, very important: I think conventions should be predictable otherwise they are not conventions (and writing rewrites sounds like something that should follow such conventions).

This means: I'd expect `afterFiles` to act the same way considering locale as `beforeFiles` as I would like both to be processing the same kind of routes - independent of their execution order.

### Which browser are you using? (if relevant)

not relevant

### How are you deploying your application? (if relevant)

not relevant

**Comment by abriginets** [comment]
Looks like the same issue as mine at #45070. Rewrites just stop working when adding `i18n` config 😕
